### PR TITLE
adding thresholded linear and rectified activation functions

### PR DESCRIPTION
**Pull request by osh**

### CODE REVIEW

File: keras/layers/advanced_activations.py [hunk 1]
```diff
@@ -36,3 +37,41 @@ def get_output(self, train):
     def get_config(self):
         return {"name":self.__class__.__name__,
         "input_shape":self.input_shape}
+
+
+class TLinear(Layer):
```
Comment (wxs) [inline keras/layers/advanced_activations.py]: There's no reason to not make these both extend `MaskedLayer` instead of `Layer` and thereby also be supported in masked scenarios.

Comment (osh) [inline keras/layers/advanced_activations.py]: agreed, pull request updated to use MaskedLayer base instead

File: docs/sources/layers/advanced_activations.md [hunk 2]
```diff
@@ -32,4 +32,40 @@ Parametrized linear unit. Similar to a LeakyReLU, where each input unit has its
...
+
+- __Input shape__: Same as `input_shape`. This layer cannot be used as first layer in a model.
+
+- __Output shape__: Same as input.
+
+- __Arguments__:
+    - __theta__: float >= 0. Threshold location of activation
+
+- __References__:
+    - [ZERO-BIAS AUTOENCODERS AND THE BENEFITS OF CO-ADAPTING FEATURES](http://arxiv.org/pdf/1402.3337.pdf)
```
Comment (fchollet) [inline docs/sources/layers/advanced_activations.md]: Please no all caps : (

File: examples/mnist_mlp_thresholdactiv.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+from keras.models import Sequential
+from keras.layers.core import Dense, Dropout, Activation
+from keras.layers.advanced_activations import TReLu, TLinear
+from keras.optimizers import SGD, Adam, RMSprop
+from keras.utils import np_utils
+
+'''
+    Train a simple deep NN on the MNIST dataset.
+
+    Get to 98.30% test accuracy after 20 epochs (there is *a lot* of margin for parameter tuning).
```
Comment (fchollet) [inline examples/mnist_mlp_thresholdactiv.py]: What is the purpose of this example? Performance is not improved compared to vanilla relu.

Comment (osh) [inline examples/mnist_mlp_thresholdactiv.py]: The purpose was really just to provide an example of how to use it and verify operation -- indeed it does not perform better at this task.  I can remove the example if you prefer.

File: docs/sources/layers/advanced_activations.md (hunk 2 above)
Comment (osh) [inline docs/sources/layers/advanced_activations.md]: fixed -- sorry this was a cut and paste thing

File: keras/layers/advanced_activations.py [hunk 4]
```diff
@@ -36,3 +37,41 @@ def get_output(self, train):
     def get_config(self):
         return {"name":self.__class__.__name__,
         "input_shape":self.input_shape}
+
+
+class TLinear(MaskedLayer):
```
Comment (iskandr) [inline keras/layers/advanced_activations.py]: Can I suggest using a longer name? Maybe `ThresholdedLinear`? It wasn't immediately obvious what `TLinear` and `TReLU` mean.

**Comment by osh** [comment]
Updated to track the latest master and follow existing style
Anything else you would like to see on this PR?

**Comment by fchollet** [comment]
Could you:
- rebase so that only changes related to the PR appear in the PR (instead of, currently, 46 different files)
- rename these two activations to more explicit names
- remove the example, as I don't think it's adding much

**Comment by osh** [comment]
Done: should be a lot cleaner now -

**Comment by osh** [comment]
Any thoughts on merging this?

**Comment by fchollet** [comment]
Let's see. I think my main issue at the time was that I couldn't immediately verify that it worked, due to the lack of test. But I'll check it out.

**Comment by fchollet** [comment]
Ok, merging!
